ndk/src/surface_texture.rs

```diff
@@ -0,0 +1,85 @@
+//! Bindings for [`ffi::ASurfaceTexture`]
+use crate::native_window::NativeWindow;
+use jni_sys::jobject;
+use jni_sys::JNIEnv;
+use std::ptr::NonNull;
+
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct SurfaceTexture {
+    ptr: NonNull<ffi::ASurfaceTexture>,
+}
+
+unsafe impl Send for SurfaceTexture {}
+
+impl Drop for SurfaceTexture {
+    fn drop(&mut self) {
+        unsafe { ffi::ASurfaceTexture_release(self.ptr.as_ptr()) }
+    }
+}
+
+impl SurfaceTexture {
+    /// Assumes ownership of `ptr`
+    ///
+    /// # Safety
+    /// `ptr` must be a valid pointer to an Android [`ffi::ASurfaceTexture`].
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::ASurfaceTexture>) -> Self {
+        Self { ptr }
+    }
+
+    pub fn ptr(&self) -> NonNull<ffi::ASurfaceTexture> {
+        self.ptr
+    }
+
+    pub fn acquire_native_window(&mut self) -> Option<NativeWindow> {
+        let native_window = unsafe { ffi::ASurfaceTexture_acquireANativeWindow(self.ptr.as_ptr()) };
+        let n = NonNull::new(native_window)?;
+        Some(unsafe { NativeWindow::from_ptr(n) })
+    }
+
+    pub fn attach_to_gl_context(&self, tex_name: u32) -> Result<(), i32> {
```

Comment:
All the `i32` results are POSIX error codes according to the documentation. We should perhaps check if the `ndk` already has an appropriate error wrapper for that or add one - even just wrapping it in a new `struct PosixError(pub i32)` seems fine by me.